src/services/accessValidation.js

```diff
@@ -24,3 +28,25 @@ export default function findOrCreateUser(data) {
       throw new Error('Error finding or creating user in database');
     }));
 }
+
+export async function validateUserAuthForAdmin(req) {
+  let result = false;
+  const { userId } = req.session;
+  const { role } = req.session;
+  let userPermissions;
+  try {
+    if (role && role === 'admin') {
+      result = true;
+    } else {
+      userPermissions = await Permission.findAll({
+        attributes: ['scopeId'],
+        where: { userId },
+      });
+      result = userPermissions.some((permission) => (permission.scopeId === 2)); // 2 is Admin
```

Comment:
Do you think it is worth it to define the scopes as constants somewhere so this would read `...scopeId === ADMIN`?